### PR TITLE
fixes according to changes in handling of FEBase

### PR DESCRIPTION
**Pull request by tarun-mitruka**
See https://github.com/ikarus-project/ikarus/pull/256 for details

### CODE REVIEW

File: src/iks003_incompressibleLinearElasticity.cpp [hunk 1]
```diff
@@ -33,55 +33,37 @@
 
 using namespace Ikarus;
 using namespace Dune::Indices;
-template <typename Basis_, typename FERequirements_ = FErequirements<>, bool useEigenRef = false>
-struct Solid {
+template <typename Basis_, typename FERequirements_ = FErequirements<>, bool useEigenRef = false, bool useFlat = true>
+struct Solid : public FEBase<Basis_, useFlat, FERequirements_, useEigenRef> {
```
Comment (rath3t) [inline src/iks003_incompressibleLinearElasticity.cpp]: Maybe for the purpose of this test it is not necessary right now to have all these templates. Solid could only be a template of the basis and FEBase<Basis_> and everything else as default. The other templates are for more advanced stuff we want to educate the users at another place.

Comment (rath3t) [reply src/iks003_incompressibleLinearElasticity.cpp]: What do you think?

File: src/iks003_incompressibleLinearElasticity.cpp [hunk 2]
```diff
@@ -33,55 +33,37 @@
...
-  LocalView &localView() { return localView_; }
-
  protected:
   template <class ScalarType>
   auto calculateScalarImpl(const FERequirementType &par, const std::optional<const Eigen::VectorX<ScalarType>> &dx
                                                          = std::nullopt) const -> ScalarType {
     const auto &d      = par.getGlobalSolution(Ikarus::FESolutions::displacement);
     const auto &lambda = par.getParameter(Ikarus::FEParameter::loadfactor);
-    Eigen::VectorX<ScalarType> localDisp(localView_.size());
+    Eigen::VectorX<ScalarType> localDisp(this->localView().size());
```
Comment (rath3t) [inline src/iks003_incompressibleLinearElasticity.cpp]: by anticipating that we want to use on the tree in the future, you can refactor this to 

```cpp
  const auto& tree = this->localView().tree();
  Eigen::VectorX<ScalarType> localDisp(tree .size());
    localDisp.setZero();
    auto &displacementNode = tree .child(_0, 0);
    auto &pressureNode     = tree .child(_1);
```

Comment (rath3t) [reply src/iks003_incompressibleLinearElasticity.cpp]: And also in the following this simplifies the example

File: src/iks004_kirchhoffPlate.cpp [hunk 3]
```diff
@@ -36,22 +35,21 @@
 #include <ikarus/utils/pythonautodiffdefinitions.hh>
 
 using namespace Ikarus;
-template <typename Basis_, typename FERequirements_ = FErequirements<>, bool useEigenRef = false>
-class KirchhoffPlate : public ScalarFieldFE<Basis_> {
+template <typename Basis_, typename FERequirements_ = FErequirements<>, bool useEigenRef = false, bool useFlat = true>
+class KirchhoffPlate : public FEBase<Basis_, useFlat, FERequirements_, useEigenRef> {
```
Comment (rath3t) [inline src/iks004_kirchhoffPlate.cpp]: Same as before wee should simplify this sohudlnt we?

**Review by rath3t — CHANGES_REQUESTED** [review]
Thanks Tarun can you think about me suggested changes? We can also chat about it tomorrow.

File: src/iks003_incompressibleLinearElasticity.cpp (hunk 1 above)
Comment (tarun-mitruka) [reply src/iks003_incompressibleLinearElasticity.cpp]: Here are my questions/thoughts that we can discuss:
1) Do we expect the users to go through the examples with the numbering system we provided (`iks001 ... iks010`) or can it be random? If it can be random, then such changes should be applied to all the examples similarly. If not, then how to decide from which example should the advanced stuff begin?
2) We should also think about who are the users and which message we want them to have as a take away. If someone wants to be a developer in the future and would like to start with learning the examples, then may be the template arguments are good to have as he/she can already know/have impression about the `class FEBase` may be. For a beginner, it is nice as `FEBase<Base_>`.
3) What will be the other place for the advanced stuffs?
4) If we simplify here, then may be we should revisit all the examples and check what else can be simplified?
In summary, I guess then my thoughts are basically to decide the purpose and scope of all the examples.

File: src/iks003_incompressibleLinearElasticity.cpp (hunk 2 above)
Comment (tarun-mitruka) [reply src/iks003_incompressibleLinearElasticity.cpp]: This is a nice idea, thanks. I think storing `tree` as a variable will simplify things in the next four instances and it will not affect the current documentation also. I will do this change after our discuss for the other comment.

In general, as we wanted to keep the PR short and particular to specific purposes, I think refactoring based on possible future changes should not be done as then converging to a particular state can take a significant amount of time.

Comment (tarun-mitruka) [reply src/iks003_incompressibleLinearElasticity.cpp]: Is this modification of storing `tree` expected also for `iks004` and `iks007`? For `iks007` may be one could also use the function `localSolutionBlockVector`, but may be these refactoring were not within the scope of this PR?

**Review by rath3t — APPROVED** [review]
